internal/consumer/service.go

```diff
@@ -0,0 +1,81 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+
+	"github.com/emmaLP/gs-software-onboarding/pkg/hackernews"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type Service struct {
+	logger          *zap.Logger
+	numberOfWorkers int
+	hnClient        hackernews.Client
+}
+
+type Client interface {
+	processStories(ctx context.Context)
+}
+
+func NewService(logger *zap.Logger, config *model.ConsumerConfig, hnClient hackernews.Client) (*Service, error) {
+	if hnClient == nil {
```

Comment:
As the `hnClient` being passed in is not a pointer, I'm not sure if it can be `null`?

Reply:
Not sure that is true. 

In the main.go file, I pass in `nil` since I just want to use the default client, but in tests I want to use a mockClient so I inject it in